File: benchmarks/accuracy/fixtures/javascript/dataflow_command_injection.js
const express = require('express');
const { exec } = require('child_process');

const router = express.Router();

router.get('/ping', (req, res) => {
    const target = req.query.host;
    const cmd = 'ping -c 1 ' + target;
    exec(cmd, (err, stdout) => res.send(stdout));
});

module.exports = router;

File: benchmarks/accuracy/fixtures/go/dataflow_command_injection.go
package main

import (
	"net/http"
	"os/exec"
)

func ping(w http.ResponseWriter, r *http.Request) {
	host := r.URL.Query().Get("host")
	cmd := "ping -c 1 " + host
	out, err := exec.Command("sh", "-c", cmd).Output()
	if err != nil {
		http.Error(w, "ping failed", http.StatusInternalServerError)
		return
	}
	w.Write(out)
}
